lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -431,18 +431,37 @@ xilinx::AIE::TileOp xilinx::AIE::ObjectFifoCreateOp::getProducerTileOp() {
 
 // ObjectFifoLinkOp
 LogicalResult xilinx::AIE::ObjectFifoLinkOp::verify() {
+  if (isJoin() && isDistribute())
+    return emitError("ObjectFifoLinkOp does not support 'join' and 'distribute' at the same time");
```

Comment:
Do you think this could be possible and useful in the future?

Reply:
I think it is possible. We'll have to think a little bit more about the interactions between the memory allocs in the DMA BDs. I don't necessarily see the use of it between shimTile -> memTile -> AIEs, but between AIEs -> memTile -> AIEs it could be useful.